lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -913,57 +924,83 @@ selectSeqStatePendingIxs ssid =
+        repsert (RndStateKey wid)
+            (RndState wid (W.getIndex (st ^. #accountIndex)))
+        insertRndStateAddresses wid sl (Rnd.addresses st)
+        cpid <- insert $ RndStateCheckpoint wid sl (st ^. #gen)
+        insertRndStatePending cpid (Rnd.pendingAddresses st)
 
     selectState (wid, sl) = runMaybeT $ do
         st <- MaybeT $ selectFirst
             [ RndStateWalletId ==. wid
+            ] []
+        stc <- MaybeT $ selectFirst
+            [ RndStateCheckpointWalletId ==. wid
             , RndStateCheckpointSlot ==. sl
             ] []
-        let (rsid, RndState _ _ accIx gen) = (entityKey st, entityVal st)
-        addresses <- lift $ selectRndStateAddresses rsid False
-        pendingAddresses <- lift $ selectRndStateAddresses rsid True
+        addresses <- lift $ selectRndStateAddresses wid sl
+        pendingAddresses <- lift $ selectRndStatePending (entityKey stc)
         rndKey <- lift $ selectRndStateKey wid
         pure $ Rnd.RndState
             { rndKey = rndKey
-            , accountIndex = W.Index accIx
+            , accountIndex = W.Index (rndStateAccountIndex (entityVal st))
             , addresses = addresses
             , pendingAddresses = pendingAddresses
-            , gen = gen
+            , gen = rndStateCheckpointGen (entityVal stc)
             }
 
     deleteState wid = do
-        rsid <- fmap entityKey <$> selectList [ RndStateWalletId ==. wid ] []
-        deleteWhere [ RndStateAddressRndStateId <-. rsid ]
+        cpid <- fmap entityKey <$> selectList [ RndStateCheckpointWalletId ==. wid ] []
+        deleteWhere [ RndStatePendingAddressCheckpointId <-. cpid ]
+        deleteWhere [ RndStateCheckpointWalletId ==. wid ]
+        deleteWhere [ RndStateAddressWalletId ==. wid ]
         deleteWhere [ RndStateWalletId ==. wid ]
 
 insertRndStateAddresses
-    :: RndStateId
-    -> Bool
+    :: W.WalletId
+    -> W.SlotId
+    -> Map (W.Index 'W.Hardened 'W.AccountK, W.Index 'W.Hardened 'W.AddressK) W.Address
+    -> SqlPersistT IO ()
+insertRndStateAddresses wid sl addresses =
+    void $ dbChunked insertMany_
+        [ RndStateAddress wid sl accIx addrIx addr
+        | ((W.Index accIx, W.Index addrIx), addr) <- Map.assocs addresses
+        ]
+
+insertRndStatePending
+    :: RndStateCheckpointId
     -> Map (W.Index 'W.Hardened 'W.AccountK, W.Index 'W.Hardened 'W.AddressK) W.Address
     -> SqlPersistT IO ()
-insertRndStateAddresses rsid pending addresses =
+insertRndStatePending cpid addresses =
     void $ dbChunked insertMany_
-        [ RndStateAddress rsid accIx addrIx addr pending
+        [ RndStatePendingAddress cpid accIx addrIx addr
         | ((W.Index accIx, W.Index addrIx), addr) <- Map.assocs addresses
         ]
 
 selectRndStateAddresses
-    :: RndStateId
-    -> Bool
+    :: W.WalletId
+    -> W.SlotId
+    -> SqlPersistT IO (Map (W.Index 'W.Hardened 'W.AccountK, W.Index 'W.Hardened 'W.AddressK) W.Address)
```

Comment:
This line (and others like it) are quite a bit longer than allowed by our coding standard.

How about the following format:
```hs
selectRndStateAddresses
    :: W.WalletId
    -> W.SlotId
    -> SqlPersistT IO
        (Map
            ( W.Index 'W.Hardened 'W.AccountK
            , W.Index 'W.Hardened 'W.AddressK )
            W.Address)
```
Admittedly it's a bit painful to have to wrap these, but it does have the advantage of showing the symmetry of pair map key type.

Alternatively, perhaps define a type synonym so that wrapping isn't necessary?

Source: https://github.com/input-output-hk/cardano-wallet/wiki/Coding-Standards#limit-line-length-to-80-characters

Reply:
I used a local type synonym.